pkgs/by-name/dw/dwlb/package.nix

```
@@ -0,0 +1,44 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, writeText
+, pkg-config
+, wayland
+, wayland-protocols
+, pixman
+, fcft
+, conf ? null
+}:
+stdenv.mkDerivation rec {
+  pname = "dwlb";
+  version = "0-unstable-2024-05-17";
+
+  src = fetchFromGitHub {
+    owner = "kolunmi";
+    repo = pname;
+    rev = "0daa1c1fdd82c4d790e477bf171e23ca2fdfa0cb";
+    sha256 = "sha256-Bu20IqRwBP1WRBgbcEQU4Q2BZ2FBnVaySOTsCn0iSSE=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ wayland wayland-protocols pixman fcft ];
+
+  outputs = [ "out" "man" ];
+
+  # Allow alternative config.def.h usage. Taken from dwl.nix.
+  postPatch = let
+    configFile = if lib.isDerivation conf || builtins.isPath conf
+                 then conf
+                 else writeText "config.def.h" conf;
+  in lib.optionalString (conf != null) "cp ${configFile} config.def.h";
+
+  installPhase = "make PREFIX=$out install";
+
+  meta = with lib; {
+    description = "A fast, feature-complete bar for dwl";
+    license = licenses.gpl3;
+    maintainers = [ maintainers.lonyelon ];
+    platforms = platforms.linux;
```

```suggestion
  meta = {
    description = "A fast, feature-complete bar for dwl";
    license = lib.licenses.gpl3;
    maintainers = with lib.maintainers; [ lonyelon ];
    platforms = platforms.linux;
```

This is a nit on my side but `with lib;` is discouraged on `meta`.

Also please add `mainProgram` field in your case it should be `mainProgram = "dwlb";`